src/lib/fragments/AgGrid.react.js

```diff
@@ -329,16 +329,14 @@ export default class DashAgGrid extends Component {
 
     onFilterChanged() {
         const {setProps, rowModelType} = this.props;
+        const virtualRowData = [];
         if (rowModelType === 'clientSide') {
-            const virtualRowData = [];
             this.state.gridApi.forEachNodeAfterFilter((node) => {
                 virtualRowData.push(node.data);
             });
-
-            const filterModel = this.state.gridApi.getFilterModel();
-            this.setState({filterModel});
-            setProps({virtualRowData});
         }
+        const filterModel = this.state.gridApi.getFilterModel();
+        setProps({virtualRowData, filterModel});
```

Comment:
Do we actually want to be setting `virtualRowData` if `rowModelType` is not `'clientSide'`? If not, we can make a `const propsToSet = {filterModel}` and inside the `if` we set `propsToSet.virtualRowData = virtualRowData`.

Reply:
Ok, yes. That's a better idea. 

I was avoiding the setProps for the virtualRowData by itself for obvious reasons, but also, it was clearing the filters too.